oecophylla/taxonomy/tests/test_parser.py

```diff
@@ -48,6 +49,22 @@ def test_extract_level(self):
         with self.assertRaisesRegex(ValueError, 'Duplicated taxa detected'):
             extract_level(table, 'p', delim='|')
 
+    def test_combine_kraken(self):
+        exp0 = pd.read_table(get_data_path('kraken/combined.tsv'),
+                             index_col=0)
+        exp0.index = exp0.index.map(str)
+        exp1 = {}
+        with open(get_data_path('kraken/lv2tids.txt'), 'r') as f:
+            for line in f:
+                lv, tids = line.rstrip('\r\n').split('\t')
+                exp1[lv] = set(tids.split(','))
+        obs0, obs1 = combine_kraken(
+            [('sampleA', get_data_path('kraken/sampleA.txt')),
+             ('sampleB', get_data_path('kraken/sampleB.txt'))])
+        assert_frame_equal(obs0[sorted(obs0.columns)].sort_index().astype(int),
+                           exp0[sorted(exp0.columns)].sort_index().astype(int))
```

Comment:
this should work fine with:

```python
        assert_frame_equal(obs0.sort_index(axis=1).astype(int),
                           exp0.sort_index(axis=1).astype(int))
```

and wouldn't this be more robust if you're casting your expectation to the type the observed value should be, rather than coercing them both to be ints in the test?

Reply:
Hi @tanaes thanks for the suggestion! I did this `astype(int)` thing because there is some auto datatype recognition issues with Pandas. BIOM is always `float`, but original read counts produced by profilers are almost always `int`. To be consistent I just converted them all to `float`.

I have confirmed that the old code works with Kraken 1.0. The current environment yaml automatically pulls the latest Kraken, which is 1.0. I am fine with forcing program versions. But it will change all programs called by oecophylla.